src/wifipaf/WiFiPAFConfig.h

```diff
@@ -0,0 +1,142 @@
+ *    simultaneous PAF connections the application will establish).
+ */
+#ifndef WIFIPAF_LAYER_NUM_PAF_ENDPOINTS
+#define WIFIPAF_LAYER_NUM_PAF_ENDPOINTS 2
+#endif // WIFIPAF_LAYER_NUM_PAF_ENDPOINTS
+
+#if (WIFIPAF_LAYER_NUM_PAF_ENDPOINTS < 1)
+#error "WIFIPAF_LAYER_NUM_PAF_ENDPOINTS must be greater than 0. configure options may be used to disable chip over PAF."
+#endif
+
+/**
+ *  @def WIFIPAF_CONNECTION_OBJECT
+ *
+ *  @brief
+ *    This defines the type of WIFIPAF_CONNECTION_OBJECT parameters passed between
+ *    PAF platform code and the WiFiPAFLayer subsystem.
+ *
+ *    Most platforms should be able to retain this type's default definition as
+ *    (void *), and pass [pointers to] connection handles generated by their
+ *    platform interface where WIFIPAF_CONNECTION_OBJECT arguments are required by
+ *    WiFiPAFLayer input functions.
+ *
+ */
+#ifndef WIFIPAF_CONNECTION_OBJECT
+#define WIFIPAF_CONNECTION_OBJECT void *
+#endif // WIFIPAF_CONNECTION_OBJECT
+
+/**
+ *  @def WIFIPAF_CONNECTION_UNINITIALIZED
+ *
+ *  @brief
+ *    This defines the value of an uninitialized WIFIPAF_CONNECTION_OBJECT.
+ *
+ */
+#ifndef WIFIPAF_CONNECTION_UNINITIALIZED
+#define WIFIPAF_CONNECTION_UNINITIALIZED nullptr
+#endif // WIFIPAF_CONNECTION_UNINITIALIZED
+
+/**
+ *  @def PAF_MAX_RECEIVE_WINDOW_SIZE
+ *
+ *  @brief
+ *    This is the maximum allowed size of a PAF end point's receive window, defined as the number of fragments the
+ *    end point may reliably receive without PAF-layer acknowledgement. This value should be no larger than the floor
+ *    of ONE-HALF the total number of slots or buffers reserved. The PAFTP layer reserves all of these buffers for its
+ *    own use - one half for incoming writes or indications, and the other half for incoming confirmations.
+ *
+ *    This value must be greater than 1, or race condition avoidance logic will prevent send the on remote device. This
+ *    logic prevents any send with no piggybacked ack when the receiver's window has only 1 slot open. Without this
+ *    logic, simultaneous data transmissions could fill both receiver's windows, leaving no room for the acks required
+ *    to re-open them. Both senders would wedge, and the PAFTP connection would stall.
+ *
+ *    This value must also exceed (WIFIPAF_CONFIG_IMMEDIATE_ACK_WINDOW_THRESHOLD + 1), or ***immediate*** stand-alone
+ *    acks will forever be sent without delay in response to one another as each peer's window size dips below
+ *    WIFIPAF_CONFIG_IMMEDIATE_ACK_WINDOW_THRESHOLD with receipt of any single message fragment.
+ *
+ *    Default value of 3 is absolute minimum for stable performance, and an attempt to ensure safe window sizes on new
+ *    platforms.
+ *
+ */
+#ifndef PAF_MAX_RECEIVE_WINDOW_SIZE
+#define PAF_MAX_RECEIVE_WINDOW_SIZE 6
+#endif
+
+#if (PAF_MAX_RECEIVE_WINDOW_SIZE < 3)
+#error "PAF_MAX_RECEIVE_WINDOW_SIZE must be greater than 2 for PAF transport protocol stability."
```

Comment:
The comments above said "greater than 1", so where does "greater than 2" come from?

Reply:
It's to add one more spare space for not sending ack so often. But, yes it's not error to set it "2". I will change it